src/main/java/edu/cornell/kfs/sys/cache/CuCompositeCacheKey.java

```diff
@@ -0,0 +1,83 @@
+package edu.cornell.kfs.sys.cache;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.kuali.kfs.sys.KFSConstants;
+
+public final class CuCompositeCacheKey {
+
+    public static final CuCompositeCacheKey EMPTY = new CuCompositeCacheKey(null, null);
+
+    private final String cacheName;
+    private final String localCacheKey;
+
+    public CuCompositeCacheKey(String cacheName, String localCacheKey) {
+        this.cacheName = StringUtils.defaultIfBlank(cacheName, KFSConstants.EMPTY_STRING);
+        this.localCacheKey = StringUtils.defaultIfBlank(localCacheKey, KFSConstants.EMPTY_STRING);
+    }
+
+    public String getCacheName() {
+        return cacheName;
+    }
+
+    public String getLocalCacheKey() {
+        return localCacheKey;
+    }
+
+    @Override
+    public boolean equals(Object otherObject) {
+        if (!(otherObject instanceof CuCompositeCacheKey)) {
+            return false;
+        }
+        CuCompositeCacheKey otherCacheKey = (CuCompositeCacheKey) otherObject;
+        return new EqualsBuilder().append(cacheName, otherCacheKey.cacheName)
+                .append(localCacheKey, otherCacheKey.localCacheKey)
+                .build();
+    }
+
+    @Override
+    public int hashCode() {
+        return new HashCodeBuilder().append(cacheName).append(localCacheKey).build();
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this).append("cacheName", cacheName)
+                .append("localCacheKey", localCacheKey)
+                .build();
+    }
+
+    public String toRedisCacheKey() {
+        return buildRedisCacheKeyInternal(cacheName, localCacheKey);
+    }
+
+    public static String buildRedisCacheKey(String cacheName, String localCacheKey) {
+        return buildRedisCacheKeyInternal(StringUtils.defaultIfBlank(cacheName, KFSConstants.EMPTY_STRING),
+                StringUtils.defaultIfBlank(localCacheKey, KFSConstants.EMPTY_STRING));
+    }
+
+    private static String buildRedisCacheKeyInternal(String cacheName, String localCacheKey) {
+        if (StringUtils.isBlank(cacheName) || StringUtils.isBlank(localCacheKey)) {
+            throw new IllegalArgumentException("Cache name and/or local key cannot be blank");
```

Comment:
would it be better to include this check in the constructor to prevent this condition early?

Reply:
This particular method returns a String and not a CuCompositeCacheKey instance. There are some areas of the code that rely upon having a CuCompositeCacheKey object that has at least a blank key, and which should have safeguards in place to prevent such objects from reaching this error condition.

As a future enhancement (or a change in this PR), we could refactor this code to help make the intent and error handling clearer.